addons/l10n_id_efaktur/models/efaktur.py

```diff
@@ -77,24 +83,30 @@ def _constrains_min_max(self):
             ], limit=1):
                 raise ValidationError(_('Efaktur interleaving range detected'))
 
-    @api.depends('min', 'max')
-    def _compute_available(self):
-        for record in self:
-            record.available = 1 + int(record.max) - int(record.min)
-
-    @api.depends('company_id')
-    def _compute_default(self):
-        for record in self:
-            query = """
-                SELECT MAX(SUBSTRING(l10n_id_tax_number FROM 4))
-                FROM account_move
-                WHERE l10n_id_tax_number IS NOT NULL
-                  AND company_id = %s
-            """
-            self.env.cr.execute(query, [record.company_id.id])
-            max_used = int(self.env.cr.fetchone()[0] or 0)
-            max_available = int(self.env['l10n_id_efaktur.efaktur.range'].search([('company_id', '=', record.company_id.id)], order='max DESC', limit=1).max)
-            record.min = record.max = '%013d' % (max(max_available, max_used) + 1)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if not vals.get('available'):
+                vals['available'] = 1 + int(vals['max']) - int(vals['min'])
```

Comment:
Do we want this if?
The system wouldn't work well if `available` isn't exactly the correct number.
I think the if can be removed and we can always set available to the expected value

Reply:
there's one exception to this case (https://github.com/odoo/odoo/blob/bdd0f7ee9e1dfec6f4269bc2cc6f5e7566ff6c6b/addons/l10n_id_efaktur/models/efaktur.py#L55) which is when we're trying to split the range and then keep the next_num of the altered range

Reply:
or I guess we can also remove if here, and swap the write/create logic when splitting

Reply:
Got it, then I guess it can say. But you can add a comment to explain why it is needed for future references.
Thank you!